faq.py

```diff
@@ -0,0 +1,114 @@
+from collections import OrderedDict
+
+CHANGELOG_QUESTION = 'What changes have been made to the test environment that might affect the data?'
+
+def get_anchor(question):
+    return question.lower().replace(' ', '-').replace('?', '').replace('\'', '')
+
+# Keys = Questions
+# Values = Markdown-supported answers
```

Comment:
If we're writing markdown, could we pull these out into a separate .md file for easier editing? Seems a bit odd and awkward to maintain markdown in Python strings?

Reply:
I agree it'd be way easier to write markdown in a md file as opposed to raw strings.

Do you have any suggestions to make that work with the autogenerated question permalinks? Best I can think of ATM is lots of string splits to get at each Q & A in a format like this file, making lots of assumptions about use of whitespace and hierarchical (h3) markdown.

Reply:
We can create a convention around each section being an h2/h3 with paragraph answer after it. 

For the slugs, did you look at existing solutions? A quick search yields..
- https://github.com/Flet/github-slugger
- https://github.com/jonschlinkert/markdown-toc

Reply:
Yeah declaring a convention (and enforcing in code reviews) would work. I'll work on a solution and ping this thread when it's ready to look at. Or LMK if it's non-blocking and I can write a TODO and file an issue.

Reply:
I wouldn't hold the launch on this, but personally I would like to see the text content living in a separate file. I'll let you make the call.

Reply:
Updated to have a separate markdown file. This is also nice because it is renderable on GitHub. See https://github.com/rviscomi/beta.httparchive.org/blob/8a8d39ae17cfcb10793d7351267d3b90e4a070f2/docs/faq.md

No other third parties were needed. I used a regex to find all headings, and map the questions to slugs. I used a similar client-side technique as the changelog to dynamically add anchors to the questions and scroll when needed. The FAQ itself is rendered server-side without the anchors built-in.